### PR TITLE
Integrate failover into into describe domain response

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Integrate failover into into describe domain response

**Why?**
To display failover info 

**How did you test it?**
Unit tests
TODO: local test

**Potential risks**
N/A

**Release notes**
Provide new field in describeDomainResponse and describe domain CLI with failover info

**Documentation Changes**

### CODE REVIEW

File: service/frontend/workflowHandler.go [hunk 1]
```diff
@@ -348,6 +348,23 @@ func (wh *WorkflowHandler) DescribeDomain(
...
+		failoverResp, err := wh.GetHistoryClient().GetFailoverInfo(ctx, &types.GetFailoverInfoRequest{
+			DomainID: resp.GetDomainInfo().UUID,
+		})
+		if err != nil {
+			// despite the error from history, return describe domain response
+			wh.GetLogger().Error(
+				fmt.Sprintf("Failed to get failover info for domain %s", resp.DomainInfo.GetName()),
+				tag.Error(err),
+			)
+			return resp, nil
```
Comment (vytautas-karpavicius) [inline service/frontend/workflowHandler.go:363]: Should we increment failure metrics as well here?

File: common/domain/handler.go [hunk 2]
```diff
@@ -345,6 +345,13 @@ func (d *handlerImpl) DescribeDomain(
 		IsGlobalDomain:  resp.IsGlobalDomain,
 		FailoverVersion: resp.FailoverVersion,
```
Comment (Shaddoll) [inline common/domain/handler.go:346]: Is this field marked as deprecated? Seems we're storing this info in FailoverInfo struct.

**Review by Shaddoll — APPROVED** [review]

File: common/domain/handler.go [hunk 3]
```diff
@@ -345,6 +345,13 @@ func (d *handlerImpl) DescribeDomain(
...
 	}
+	if resp.FailoverEndTime != nil {
+		response.FailoverInfo = &types.FailoverInfo{
+			FailoverVersion:         resp.FailoverVersion,
+			FailoverStartTimestamp:  resp.LastUpdatedTime,
```
[the remark this answers is not in the document]
Comment (yux0) [reply common/domain/handler.go:351]: Yes. This is correct. This is to give an idea of when the failover is started.

File: service/frontend/workflowHandler.go (hunk 1 above)
Comment (yux0) [reply service/frontend/workflowHandler.go:363]: We can track it in the GetFailoverInfo API. The describe domain API will return the response from domain table

**Review by yycptt — APPROVED** [review]

**Review by vytautas-karpavicius — APPROVED** [review]

**Pull request merged**